### PR TITLE
1138 bal radio group add prop options like the new steps

### LINKED ISSUE #1138: [bal-radio-group] -  add prop options like the new steps
Before
```html
<bal-radio-group value="1">
    <bal-radio name="radio-example" value="1">Label 1</bal-radio>
    <bal-radio name="radio-example" value="2">Label 2</bal-radio>
    <bal-radio name="radio-example" value="3" disabled>Disabled</bal-radio>
    <bal-radio name="radio-example" value="4">Random text with a <a class="is-link" target="_blank" href="http://baloise.ch">Link</a> in it</bal-radio>
</bal-radio-group>
```

After
```html
<bal-radio-group value="1" options={options}></bal-radio-group>
```

```JavaScript
const options = [
newBalRadioOption({ name:"...", label:"",  value:"" }),
newBalRadioOption({ name:"...", label:"",  value:"" }),
...
]
```

### PR DESCRIPTION
**Pull request by mladenplaninicic**
Closes #

{{short description}}

## Acceptance Criteria

- Design Review by @Gagne87, @clastzoo
- Technical Review by @hirsch88, @mladenplaninicic
- [Style Guide](https://stenciljs.com/docs/style-guide) Review @hirsch88, @mladenplaninicic
- Visual test done by @hirsch88, @mladenplaninicic, @Gagne87, @clastzoo
- Functional test done by @hirsch88, @mladenplaninicic
- Browser Testing by @hirsch88, @mladenplaninicic, @Gagne87, @clastzoo
    - Desktop
        - Chrome
        - Edge
        - Safari
        - Firefox
    - Tablet
        - iPad (Landscape / Portrait)
    - Mobile
        - Safari iOS
        - Chrome Preview

### CODE REVIEW

File: packages/components/src/components/form/bal-radio/bal-radio-group/bal-radio-group.tsx [hunk 1]
```diff
@@ -158,7 +172,19 @@ export class RadioGroup implements ComponentInterface, Loggable {
   }
 
   componentDidLoad() {
-    this.setRadioTabindex(this.value)
+    this.onOptionChange()
+    this.mutationO = watchForRadios<HTMLBalRadioElement>(this.el, 'bal-step-item', () => {
```
Comment (hirsch88) [inline packages/components/src/components/form/bal-radio/bal-radio-group/bal-radio-group.tsx:176]: - [x] why do you add baa-step-item?

File: packages/components/src/components/form/bal-radio/bal-radio-group/bal-radio-group.tsx [hunk 2]
```diff
@@ -319,6 +357,14 @@ export class RadioGroup implements ComponentInterface, Loggable {
     }
   }
 
+  private onOptionChange = async () => {
+    try {
+      this.setRadioTabindex(this.value)
+    } catch (e) {
+      console.warn('[WARN] - Could not read radio options')
```
Comment (hirsch88) [inline packages/components/src/components/form/bal-radio/bal-radio-group/bal-radio-group.tsx:364]: - [x] I think we should fail silently or what do you think? if there are none we render none :-)

File: packages/components/src/components/form/bal-radio/bal-radio.tsx [hunk 3]
```diff
@@ -201,6 +202,30 @@ export class Radio implements ComponentInterface, Loggable {
     return Promise.resolve(this.nativeInput)
   }
 
+  /**
+   * Options of the tab like label, value etc.
+   */
+  @Method()
+  async getOptions(): Promise<BalRadioOption> {
```
Comment (hirsch88) [inline packages/components/src/components/form/bal-radio/bal-radio.tsx:209]: - [x] use singular not plural

File: packages/components/src/components/form/bal-radio/test/bal-radio.cy.html [hunk 4]
```diff
@@ -69,6 +69,42 @@
...
+
+        <bal-heading>Options</bal-heading>
+        <section>
+          <bal-radio-group id="options-test" data-testid="options" value="1"></bal-radio-group>
+          <script>
+            const stepEl = document.getElementById('options-test')
+            stepEl.options = [
+              {
+                value: '1',
+                name: 'test',
```
Comment (hirsch88) [inline packages/components/src/components/form/bal-radio/test/bal-radio.cy.html:81]: - [x] only pass required fields

File: packages/components/src/components/form/bal-radio/utils/watch-radios.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
+export const watchForRadios = <T extends HTMLElement>(
```
Comment (hirsch88) [inline packages/components/src/components/form/bal-radio/utils/watch-radios.ts:1]: - [x] create a util out of that because we use that every where

**Review by hirsch88 — CHANGES_REQUESTED** [review]

**Comment by mladenplaninicic** [comment]
@hirsch88 issues are fixed and this branch is merged in checkbox-radio :)

**Pull request closed without merging**